kernel/nativecode.ml

```diff
@@ -1363,12 +1396,12 @@ let compile_prim env decl cond paux =
       | None -> MLarray [||]
       | Some u -> MLlocal u
     in
-    let uarg = MLapp(MLprimitive MLmagic, [|uarg|]) in
-    MLapp(MLprimitive Mk_sort, [|get_sort_code i; uarg|])
+    (* FIXME: use a dedicated cast function *)
```

Comment:
What needs to be done here?

Reply:
These lines are playing with fire w.r.t. type-checking. Basically we're sending Coq-side data to the native compilation scheme as a native value `t` type, but this is not an actual native value. It doesn't really matter in the end because we're morally untyped, but it could be problematic if we tried to preserve more invariants. (This FIXME is not about a behaviour introduced by this PR, it's already latent in the master code.)